core/src/main/java/org/axonframework/eventhandling/scheduling/quartz/QuartzEventScheduler.java

```diff
@@ -192,27 +212,111 @@ public void setEventJobDataBinder(EventJobDataBinder jobDataBinder) {
          */
+        @Deprecated
         public static final String EVENT_KEY = EventMessage.class.getName();
 
+        private final Serializer serializer;
+
+        /**
+         * Instantiate a {@link DirectEventJobDataBinder} which defaults to a {@link XStreamSerializer} for
+         * de-/serializing event messages.
+         */
+        public DirectEventJobDataBinder() {
+            this(new XStreamSerializer());
+        }
+
+        /**
+         * Instantiate a {@link DirectEventJobDataBinder} with the provided {@link Serializer} for
+         * de-/serializing event messages.
+         *
+         * @param serializer the {@link Serializer} used for de-/serializing event messages
+         */
+        public DirectEventJobDataBinder(Serializer serializer) {
+            this.serializer = serializer;
+        }
+
         @Override
-        public JobDataMap toJobData(Object eventMessage) {
+        public JobDataMap toJobData(Object event) {
             JobDataMap jobData = new JobDataMap();
-            jobData.put(EVENT_KEY, eventMessage);
+
+            EventMessage eventMessage = (EventMessage) event;
+
+            jobData.put(MESSAGE_ID, eventMessage.getIdentifier());
+            jobData.put(MESSAGE_TIMESTAMP, eventMessage.getTimestamp().toEpochMilli());
+
+            SerializedObject<byte[]> serializedPayload =
+                    serializer.serialize(eventMessage.getPayload(), byte[].class);
+            jobData.put(SERIALIZED_MESSAGE_PAYLOAD, serializedPayload.getData());
+            jobData.put(MESSAGE_TYPE, serializedPayload.getType().getName());
+            jobData.put(MESSAGE_REVISION, serializedPayload.getType().getRevision());
+
+            SerializedObject<byte[]> serializedMetaData =
+                    serializer.serialize(eventMessage.getMetaData(), byte[].class);
+            jobData.put(MESSAGE_METADATA, serializedMetaData.getData());
+
             return jobData;
         }
 
+        /**
+         * <b>Note</b> that this function is able to retrieve an Event Message in two formats. The first is the
+         * original, deprecated approach, which uses the Quartz serializer (read Java serialization) to serialize an
+         * entire event message at once. The second approach uses the configurable {@link Serializer} in this class to
+         * serialized the payload and metadata of an event message. All fields which required to instantiate a
+         * {@link GenericEventMessage} are currently stored in the {@code jobDataMap} when calling the
+         * {@link DirectEventJobDataBinder#toJobData(Object)} function.
+         * Approach one only exists for backwards compatibility and should be removed in subsequent major releases.
+         * <p>
+         * {@inheritDoc}
+         *
+         * @deprecated the if-block should be removed on the next major release | it is only maintained for backwards
+         * compatibility
+         */
+        @Deprecated
```

Comment:
This method is going to be used in a new version also. The thing that is deprecated here is checking serialized content. I wouldn't mark the whole method as deprecated.

Reply:
Was doubting how to document this to begin with.
Made some changes to the approach now, please share your thoughts on them once you've got the time.